test/cql-pytest/test_aggregate.py

```diff
@@ -72,17 +72,24 @@ def test_count_in_partition(cql, table1):
     cql.execute(stmt, [p, 3, 3])
     assert [(3,)] == list(cql.execute(f"select count(*) from {table1} where p = {p}"))
 
-# Using count(v) instead of count(*) allows counting only rows with a set
+# Using count(v) instead of count(*) allows counting only rows with a non-NULL
 # value in v
-def test_count_specific_column(cql, table1):
+def test_count_specific_column(cql, test_keyspace, table1):
     p = unique_key_int()
     stmt = cql.prepare(f"insert into {table1} (p, c, v) values (?, ?, ?)")
     cql.execute(stmt, [p, 1, 1])
     cql.execute(stmt, [p, 2, 2])
-    cql.execute(stmt, [p, 3, 3])
+    cql.execute(stmt, [p, 3, 3,])
     cql.execute(stmt, [p, 4, None])
     assert [(4,)] == list(cql.execute(f"select count(*) from {table1} where p = {p}"))
     assert [(3,)] == list(cql.execute(f"select count(v) from {table1} where p = {p}"))
+    # Check with non-scalar types too, reproduces #14198
+    cql.execute(f'CREATE TYPE {test_keyspace}.my_udt (i int, t text)');
```

Comment:
The benefits of the `new_type()` context manager over this command:
1. It picks a random name instead of "my_udt". If another one of the 700 tests we have uses the same name, it will fail.
2. It remembers to delete the type after the test is over (you forgot), and also if the test fails in the middle.

Concretetly, running this test twice (test/cql-pytest/run --count=2 test/cql-pytest/test_aggregate.py::the_test_name) would fail because it can't create the already-existing type name.